### DISCUSSION TITLE
Subscription Resolver - Date Serialization

### DISCUSSION THREAD

**Discussion by Falco-Boehnke** [MichalLytek/typegraphql-prisma#215] [Q&A]
Hi there.

I added a scubscription resolver to my schema, I call it via a middleware that publishes the topic. I filter which model published that notification, then the subscription is called. Now [this github issue #600 in typegraphql](https://github.com/MichalLytek/type-graphql/issues/600) popped up in my code.

Using DateTimeResolver from the graphql-scalars package seems to be solving the issue.

```
import { DateTimeResolver } from 'graphql-scalars';

@Field(() => DateTimeResolver)
```

Is it possible to adapt my current schema:
```
model Project {
    // Basic
    id          String   @id @default(uuid())
    createdAt   DateTime @default(now())
    updatedAt   DateTime @updatedAt
}
```

into using that DateTimeResolver type on that typegraphql field? 

```
model Project {
    // Basic
    id          String   @id @default(uuid())
    
    /// @TypeGraphQL.field(type => GraphQLScalars.DateTimeResolver)
    createdAt   DateTime @default(now())
   
    /// @TypeGraphQL.field(type => GraphQLScalars.DateTimeResolver)
    updatedAt   DateTime @updatedAt
}
```

Custom Decorators seem to be one other possible solution to this, although I have not managed to defined a type in them (couldn't figure out how, it at all possible.)

```
applyModelsEnhanceMap({
    Project: {
        fields: {
           // How to tell the customizer the field type?
            createdAt: [],
        },
    },
});

```

If neither option is feasible, I will just adapt my SubscriptionPublisherMiddleware to parse the DateTime fields, but that feels... wrong. Any tip is greatly appreciated!

**Comment by MichalLytek** [comment]
> I will just adapt my SubscriptionPublisherMiddleware to parse the DateTime fields, but that feels... wrong

It feels a correct approach. Imagine you call another service to process the published notification, your model says `Date` but in real it's `string` so it throws `TypeError` while calling `value.toISOString()` 😉 

If your underlying pubsub engine is serializing `Date`, `BigInt`, etc. you should reparse that back to the proper JS types.
This ways it will be synced with types definition and the Date scalar will work correctly.
